tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/TypeSpec/CustomizedCodeUpdateTool.cs

```diff
@@ -184,82 +185,117 @@ private async Task<CustomizedCodeUpdateResponse> UpdateCoreAsync(string commitSh
             return (false, ex.Message);
         }
     }
-    private async Task<(bool Success, bool RequiresReview)> ValidateAndUpdateGuidanceAsync(
-        string packagePath,
-        LanguageService languageService,
-        List<string> guidance,
-        CancellationToken ct)
-    {
-        var validationResult = await languageService.ValidateAsync(packagePath, ct);
 
-        if (validationResult.Success)
+
+    /// <summary>
+    /// Builds the SDK project to validate compilation.
+    /// </summary>
+    private async Task<(bool Success, string? ErrorMessage)> BuildSdkAsync(string packagePath, LanguageService languageService, CancellationToken ct)
+    {
+        try
```

Comment:
@raych1 I know you were thinking about ways to potentially share this with the BuildTool did we come up with a plan to share?

Reply:
Yes, I discussed with Sameeksha offline and agreed to duplicate this code for now. The plan is to move the common code to the language service. This is [the issue](https://github.com/Azure/azure-sdk-tools/issues/12852) to track the work.